src/protocol/attribution/input.rs

```diff
@@ -0,0 +1,172 @@
+use crate::bits::BitArray;
+use crate::error::Error;
+use crate::ff::Field;
+use crate::helpers::Role;
+use crate::protocol::context::Context;
+use crate::protocol::sort::apply_sort::shuffle::Resharable;
+use crate::protocol::{RecordId, Substep};
+use crate::secret_sharing::replicated::semi_honest::{AdditiveShare, XorShare};
+use async_trait::async_trait;
+use futures::future::{try_join, try_join_all};
+
+//
+// `accumulate_credit` protocol
+//
+#[derive(Debug)]
+pub struct AccumulateCreditInputRow<F: Field, BK: BitArray> {
+    pub is_trigger_report: AdditiveShare<F>,
+    pub helper_bit: AdditiveShare<F>,
+    pub breakdown_key: XorShare<BK>,
+    pub trigger_value: AdditiveShare<F>,
+}
+
+#[derive(Debug)]
+pub struct MCAccumulateCreditInputRow<F: Field> {
```

Comment:
What does "MCA" stand for? I don't quite get this name.

Reply:
MC = Modulus Converted. Names getting too long, it might be a time we need to use abbreviations.